packages/fx-core/src/plugins/resource/aad/utils/configs.ts

```diff
@@ -267,7 +268,7 @@ export class ProvisionConfig {
     inputs: v2.InputsWithProjectPath,
     localSettings: v2.LocalSettings
   ): Promise<Result<any, FxError>> {
-    const displayName: string = ctx.projectSetting.appName;
+    const displayName: string = convertToAlphanumericOnly(ctx.projectSetting.appName);
```

Comment:
Seems multiple places has these code, consider extract to a function:
```
    const displayName: string = convertToAlphanumericOnly(ctx.projectSetting.appName);
    if (displayName) {
      this.displayName = displayName.substr(0, Constants.aadAppMaxLength) as string;
    } else {
      throw ResultFactory.SystemError(
        GetConfigError.name,
        GetConfigError.message(Errors.GetDisplayNameError[0])
      );
    }
```

Reply:
Fixed